lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -110,11 +109,34 @@ update the C<accounts> variable every 10 seconds
 async sub update_accounts {
     my $self = shift;
     while (1) {
+        await $self->get_hash_accounts();
         await $self->loop->delay_future(after => UPDATE_ACCOUNTS);
-        $self->{accounts} = await $self->rpc_client->accounts();
     }
 }
 
+=head2 get_hash_accounts
+
+Request the node accounts and convert it to a hash
+
+=over 4
+
+=back
+
+hash ref containing the accounts as keys
+
+=cut
+
+async sub get_hash_accounts {
+    my ($self) = @_;
+
+    my $accounts_response = await $self->rpc_client->accounts();
+    my %accounts = map { lc($_) => 1 } $accounts_response->@*;
+    $self->{accounts} = \%accounts;
```

Comment:
Does this work?  seems to me  you make a reference to (not a copy) of %accounts then undef %accounts which would undef $self->{accounts}.  
```
$ my %hash = (a =>1, b=>2)
$VAR1 = 'a';
$VAR2 = 1;
$VAR3 = 'b';
$VAR4 = 2;

$ my $hash_ref = \%hash
$HASH1 = {
           a => 1,
           b => 2
         };

$ undef %hash

$ $hash_ref
$HASH1 = {};
```

Reply:
Thanks, yes, this was setting it as `undef`, I removed the `undef` and added some more tests to this case.